src/chainparams.cpp

```diff
@@ -339,6 +357,25 @@ const CChainParams &Params() {
     return *globalChainParams;
 }
 
+CChainParams& Params(const std::string& chain)
```

Comment:
You're right, this function should not be needed and I will re-use CreateChainParams.
The params created in Params() are not explicitly destroyed.